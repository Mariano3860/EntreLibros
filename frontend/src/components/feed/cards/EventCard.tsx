import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { EventItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: EventItem
}

export const EventCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handleGo = () => {
    track('feed.cta', { type: 'event', action: 'go' })
  }

  const image = `https://picsum.photos/seed/${item.title}/600/400`

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <img src={image} alt={item.title} className={styles.image} />
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <p>
          {item.date} Â· {item.location}
        </p>
        <button
          className={styles.primaryButton}
          onClick={handleGo}
          aria-label={t('community.feed.cta.go')}
        >
          {t('community.feed.cta.go')}
        </button>
      </div>
    </article>
  )
}
