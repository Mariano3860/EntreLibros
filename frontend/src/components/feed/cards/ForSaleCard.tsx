import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { SaleItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: SaleItem
}

export const ForSaleCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handleBuy = () => {
    track('feed.cta', { type: 'sale', action: 'buy' })
  }

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <img src={item.cover} alt={item.title} className={styles.image} />
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <p>{t('community.feed.sale.price', { price: item.price })}</p>
        <button
          className={styles.primaryButton}
          onClick={handleBuy}
          aria-label={t('community.feed.cta.buy')}
        >
          {t('community.feed.cta.buy')}
        </button>
      </div>
    </article>
  )
}
