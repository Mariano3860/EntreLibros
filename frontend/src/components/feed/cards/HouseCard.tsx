import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { HouseItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: HouseItem
}

export const HouseCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handleMap = () => {
    track('feed.cta', { type: 'house', action: 'map' })
  }

  const image = `https://picsum.photos/seed/${item.name}/600/400`

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <img src={image} alt={item.name} className={styles.image} />
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <h3 className={styles.title}>{item.name}</h3>
        <p>{t('community.feed.house.distance', { distance: item.distance })}</p>
        <button
          className={styles.secondaryButton}
          onClick={handleMap}
          aria-label={t('community.feed.cta.open_map')}
        >
          {t('community.feed.cta.open_map')}
        </button>
      </div>
    </article>
  )
}
