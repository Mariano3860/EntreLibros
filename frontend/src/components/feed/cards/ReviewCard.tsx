import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { ReviewItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: ReviewItem
}

export const ReviewCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handleHelpful = () => {
    track('feed.cta', { type: 'review', action: 'helpful' })
  }

  const image = `https://picsum.photos/seed/${item.book}/600/400`

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <img src={image} alt={item.book} className={styles.image} />
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <h3 className={styles.title}>{item.book}</h3>
        <p>{item.quote}</p>
        <button
          className={styles.secondaryButton}
          onClick={handleHelpful}
          aria-label={t('community.feed.cta.helpful')}
        >
          {t('community.feed.cta.helpful')}
        </button>
      </div>
    </article>
  )
}
