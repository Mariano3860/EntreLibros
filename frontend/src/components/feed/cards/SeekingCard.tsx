import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { SeekingItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: SeekingItem
}

export const SeekingCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handleOffer = () => {
    track('feed.cta', { type: 'seeking', action: 'offer' })
  }

  const image = `https://picsum.photos/seed/${item.title}/600/400`

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <img src={image} alt={item.title} className={styles.image} />
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <p>
          {t('community.feed.seeking.description', {
            user: item.user,
            title: item.title,
          })}
        </p>
        <button
          className={styles.primaryButton}
          onClick={handleOffer}
          aria-label={t('community.feed.cta.offer_copy')}
        >
          {t('community.feed.cta.offer_copy')}
        </button>
      </div>
    </article>
  )
}
