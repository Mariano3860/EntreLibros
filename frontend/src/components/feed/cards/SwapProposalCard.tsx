import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import { FeedActions } from '../FeedActions'
import type { SwapProposalItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'
import { FeedCardHeader } from './FeedCardHeader'

interface Props {
  item: SwapProposalItem
}

export const SwapProposalCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const offeredImage =
    item.offered.cover ??
    `https://picsum.photos/seed/${item.offered.id}/600/400`
  const requestedImage =
    item.requested.cover ??
    `https://picsum.photos/seed/${item.requested.id}/600/400`

  const handleAccept = () => {
    track('feed.cta', { type: 'swap', action: 'accept' })
  }

  return (
    <article className={styles.card}>
      <FeedCardHeader item={item} />
      <div className={styles.swapImages}>
        <img src={offeredImage} alt={item.offered.title} />
        <img src={requestedImage} alt={item.requested.title} />
      </div>
      <FeedActions initialLikes={item.likes} />
      <div className={styles.content}>
        <section
          aria-label={t('community.feed.swap.proposalAria', {
            user: item.requester.displayName,
          })}
        >
          <p>
            {t('community.feed.swap.wants', {
              requester: item.requester.displayName,
              offeredTitle: item.offered.title,
              requestedTitle: item.requested.title,
              requestedOwner: item.requested.owner.displayName,
            })}
          </p>
          <dl className={styles.srOnly}>
            <dt>{t('community.feed.swap.requester')}</dt>
            <dd>{item.requester.displayName}</dd>
            <dt>{t('community.feed.swap.offered')}</dt>
            <dd>
              {item.offered.title}
              {item.offered.author ? ` — ${item.offered.author}` : ''}
            </dd>
            <dt>{t('community.feed.swap.requested')}</dt>
            <dd>
              {item.requested.title}
              {item.requested.author ? ` — ${item.requested.author}` : ''}
            </dd>
            <dt>{t('community.feed.swap.requestedOwner')}</dt>
            <dd>{item.requested.owner.displayName}</dd>
          </dl>
        </section>
        <button
          className={styles.primaryButton}
          onClick={handleAccept}
          aria-label={t('community.feed.cta.accept')}
        >
          {t('community.feed.cta.accept')}
        </button>
      </div>
    </article>
  )
}
