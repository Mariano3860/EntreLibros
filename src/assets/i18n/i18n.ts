import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'

import enTranslations from './locales/en/common.json'
import esTranslations from './locales/es/common.json'

declare module 'react-i18next' {
  interface Resources {
    translation: typeof esTranslations
  }
}

i18n.use(initReactI18next).init({
  resources: {
    en: { translation: enTranslations },
    es: { translation: esTranslations },
  },
  lng: 'es',
  fallbackLng: 'es',
  interpolation: {
    escapeValue: false,
  },
})

export default i18n
