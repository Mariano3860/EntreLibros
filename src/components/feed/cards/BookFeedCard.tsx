import { useTranslation } from 'react-i18next'

import { track } from '@src/utils/analytics'

import type { BookItem } from '../FeedItem.types'

import styles from './FeedCard.module.scss'

interface Props {
  item: BookItem
}

export const BookFeedCard = ({ item }: Props) => {
  const { t } = useTranslation()

  const handlePrimary = () => {
    track('feed.cta', { type: 'book', action: 'primary' })
  }

  return (
    <article className={styles.card}>
      <h3 className={styles.title}>{item.title}</h3>
      <p>{item.author}</p>
      <div className={styles.actions}>
        <button
          className={styles.primaryButton}
          onClick={handlePrimary}
          aria-label={t('community.feed.cta.propose_swap')}
        >
          {t('community.feed.cta.propose_swap')}
        </button>
      </div>
    </article>
  )
}
