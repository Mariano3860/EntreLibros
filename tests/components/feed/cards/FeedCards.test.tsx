import { describe, expect, test } from 'vitest'

import i18n from '@src/assets/i18n/i18n'
import { BookFeedCard } from '@src/components/feed/cards/BookFeedCard'
import { EventCard } from '@src/components/feed/cards/EventCard'
import { ForSaleCard } from '@src/components/feed/cards/ForSaleCard'
import { HouseCard } from '@src/components/feed/cards/HouseCard'
import { PeopleCard } from '@src/components/feed/cards/PeopleCard'
import { ReviewCard } from '@src/components/feed/cards/ReviewCard'
import { SeekingCard } from '@src/components/feed/cards/SeekingCard'
import { SwapProposalCard } from '@src/components/feed/cards/SwapProposalCard'
import type {
  BookItem,
  EventItem,
  HouseItem,
  PersonItem,
  ReviewItem,
  SaleItem,
  SeekingItem,
  SwapProposalItem,
} from '@src/components/feed/FeedItem.types'

import { renderWithProviders } from '../../../test-utils'

const base = {
  user: 'User',
  avatar: 'https://i.pravatar.cc/100?img=1',
  time: 'now',
  likes: 0,
}

const book: BookItem = {
  ...base,
  type: 'book',
  id: 'b',
  title: 'Dune',
  author: 'Frank Herbert',
  cover: 'https://picsum.photos/seed/dune/600/400',
}

const swap: SwapProposalItem = {
  ...base,
  type: 'swap',
  id: 's',
  requester: 'Ana',
  offered: '1984',
  requested: 'The Hobbit',
}

const sale: SaleItem = {
  ...base,
  type: 'sale',
  id: 'sa',
  title: 'Foundation',
  price: 10,
  condition: 'used',
  cover: 'https://picsum.photos/seed/foundation/600/400',
}

const seeking: SeekingItem = {
  ...base,
  type: 'seeking',
  id: 'se',
  title: 'Sapiens',
}

const review: ReviewItem = {
  ...base,
  type: 'review',
  id: 'r',
  book: 'Dune',
  quote: 'Great book',
}

const event: EventItem = {
  ...base,
  type: 'event',
  id: 'e',
  title: 'Book Fair',
  date: '2025-01-01',
  location: 'Casita Centro',
}

const house: HouseItem = {
  ...base,
  type: 'house',
  id: 'h',
  name: 'Casita Norte',
  distance: 1.2,
}

const person: PersonItem = {
  ...base,
  type: 'person',
  id: 'p',
  name: 'Carlos',
  match: 80,
  lastActivity: 'yesterday',
}

const cases = [
  ['book', <BookFeedCard item={book} />],
  ['swap', <SwapProposalCard item={swap} />],
  ['sale', <ForSaleCard item={sale} />],
  ['seeking', <SeekingCard item={seeking} />],
  ['review', <ReviewCard item={review} />],
  ['event', <EventCard item={event} />],
  ['house', <HouseCard item={house} />],
  ['person', <PeopleCard item={person} />],
] as const

describe('Feed cards snapshots', () => {
  test.each(cases)('%s card matches snapshot (light/en)', async (_name, element) => {
    localStorage.setItem('theme', 'light')
    await i18n.changeLanguage('en')
    const { asFragment } = renderWithProviders(element)
    expect(asFragment()).toMatchSnapshot()
    localStorage.clear()
  })

  test.each(cases)('%s card matches snapshot (dark/es)', async (_name, element) => {
    localStorage.setItem('theme', 'dark')
    await i18n.changeLanguage('es')
    const { asFragment } = renderWithProviders(element)
    expect(asFragment()).toMatchSnapshot()
    localStorage.clear()
  })
})
