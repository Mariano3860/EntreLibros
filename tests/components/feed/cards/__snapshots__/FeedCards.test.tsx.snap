// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Feed cards snapshots > book card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Dune
    </h3>
    <p>
      Frank Herbert
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.propose_swap"
      >
        community.feed.cta.propose_swap
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > book card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Dune
    </h3>
    <p>
      Frank Herbert
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.propose_swap"
      >
        community.feed.cta.propose_swap
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > event card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Book Fair
    </h3>
    <p>
      Casita Centro
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.go"
      >
        community.feed.cta.go
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > event card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Book Fair
    </h3>
    <p>
      Casita Centro
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.go"
      >
        community.feed.cta.go
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > house card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Casita Norte
    </h3>
    <p>
      community.feed.house.distance
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.open_map"
      >
        community.feed.cta.open_map
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > house card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Casita Norte
    </h3>
    <p>
      community.feed.house.distance
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.open_map"
      >
        community.feed.cta.open_map
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > person card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Carlos
    </h3>
    <p>
      community.feed.person.match
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.message"
      >
        community.feed.cta.message
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > person card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Carlos
    </h3>
    <p>
      community.feed.person.match
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.message"
      >
        community.feed.cta.message
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > review card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Maria
    </h3>
    <p>
      Great book
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.helpful"
      >
        community.feed.cta.helpful
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > review card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Maria
    </h3>
    <p>
      Great book
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.helpful"
      >
        community.feed.cta.helpful
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > sale card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Foundation
    </h3>
    <p>
      community.feed.sale.price
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.buy"
      >
        community.feed.cta.buy
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > sale card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Foundation
    </h3>
    <p>
      community.feed.sale.price
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.buy"
      >
        community.feed.cta.buy
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > seeking card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Luis
    </h3>
    <p>
      community.feed.seeking.description
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.offer_copy"
      >
        community.feed.cta.offer_copy
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > seeking card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Luis
    </h3>
    <p>
      community.feed.seeking.description
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.offer_copy"
      >
        community.feed.cta.offer_copy
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > swap card matches snapshots 1`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Ana
    </h3>
    <p>
      Ana community.feed.swap.wants
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.accept"
      >
        community.feed.cta.accept
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Feed cards snapshots > swap card matches snapshots 2`] = `
<DocumentFragment>
  <article
    class="_card_9f5b00"
  >
    <h3
      class="_title_9f5b00"
    >
      Ana
    </h3>
    <p>
      Ana community.feed.swap.wants
    </p>
    <div
      class="_actions_9f5b00"
    >
      <button
        aria-label="community.feed.cta.accept"
      >
        community.feed.cta.accept
      </button>
    </div>
  </article>
</DocumentFragment>
`;
